crates/re_space_view_tensor/src/space_view_class.rs

```diff
@@ -152,21 +137,41 @@ impl SpaceViewClass for TensorSpaceView {
             view_property_ui::<TensorViewFit>(ctx, ui, view_id, self, state);
         });
 
+        // TODO(andreas): Listitemify
         if let Some((_, tensor)) = &state.tensor {
+            let slice_property = ViewProperty::from_archetype::<TensorSliceSelection>(
+                ctx.blueprint_db(),
+                ctx.blueprint_query,
+                view_id,
+            );
+            let slice_selection =
+                load_tensor_slice_selection_and_make_valid(&slice_property, tensor.shape())?;
+
             ui.separator();
             ui.strong("Dimension Mapping");
-            dimension_mapping_ui(ui, &mut state.slice.dim_mapping, tensor.shape());
-            let default_mapping = DimensionMapping::create(tensor.shape());
+            dimension_mapping_ui(ctx, ui, tensor.shape(), &slice_selection, &slice_property);
+
+            // TODO(andreas): this is a bit too inconsistent with the other UIs.
             if ui
-                .add_enabled(
-                    state.slice.dim_mapping != default_mapping,
-                    egui::Button::new("Reset mapping"),
-                )
-                .on_disabled_hover_text("The default is already set up")
-                .on_hover_text("Reset dimension mapping to the default")
+                .button("Reset to default blueprint")
+                .on_hover_text("Reset dimension mapping to the previously set default blueprint")
+                .clicked()
+            {
+                slice_property.reset_all_components(ctx);
+            }
+
+            if ui
+                .add_enabled_ui(slice_property.any_non_empty(), |ui| {
+                    ui.button("Reset to default")
+                        .on_hover_text(
+                            "Reset dimension mapping to the default, i.e. as if never set",
+                        )
+                        .on_disabled_hover_text("No custom dimension mapping set")
+                })
+                .inner
                 .clicked()
```

Comment:
There is a shorter way of doing this:

```suggestion
            if ui.add_enabled(slice_property.any_non_empty(), egui::Button::new("Reset to default"))
                .on_hover_text("Reset dimension mapping to the default, i.e. as if never set")
                .on_disabled_hover_text("No custom dimension mapping set")
                .clicked()
```